crates/e2e/src/setup/solver.rs

```diff
@@ -0,0 +1,245 @@
+//! Mock solver for testing purposes. It returns a custom solution.
+
+use {
+    app_data::AppDataHash,
+    axum::Json,
+    ethcontract::{common::abi::ethereum_types::Address, jsonrpc::serde::Serialize},
+    model::{
+        order::{BuyTokenDestination, OrderData, OrderKind, SellTokenSource},
+        signature::EcdsaSigningScheme,
+        DomainSeparator,
+    },
+    solvers_dto::{
+        auction::Auction,
+        solution::{Asset, Kind, Solution, Solutions, Trade},
+    },
+    std::{future::Future, net::SocketAddr, sync::Arc},
+    tokio::{
+        signal::{unix, unix::SignalKind},
+        sync::oneshot,
+    },
+    tracing::Instrument,
+    warp::hyper,
+    web3::signing::SecretKeyRef,
+};
+
+pub struct Mock {
+    solution: Solution,
+}
+
+impl Mock {
+    pub fn new(solution: Solution) -> Self {
+        Self { solution }
+    }
+
+    /// Returns the specified solution with the same order UID as the order UID
+    /// in the auction.
+    pub async fn solve(&self, auction: Auction) -> Solutions {
+        let mut solution = self.solution.clone();
+        // Return the same order UID as the order UID in the auction
+        solution
+            .trades
+            .iter_mut()
+            .filter_map(|trade| match trade {
+                Trade::Fulfillment(fulfillment) => Some(fulfillment),
+                Trade::Jit(_) => None,
+            })
+            .zip(auction.orders.iter())
+            .for_each(|(fulfillment, order)| fulfillment.order = order.uid);
```

Comment:
I think the solver should be as dumb as possible and only return exactly what was configured in the test. The test just has to be structured such that you know the order uids before spawning the solver. Which should be okay because you can spawn the orderbook before the autopilot and the driver for the mock solver. (would require 2 separate drivers which might have inspired this workaround?).